packages/runtime/container-runtime/README.md

```diff
@@ -51,6 +51,52 @@ issue](https://github.com/microsoft/FluidFramework/wiki/Submitting-Bugs-and-Feat
 
 Thank you!
 
+## Data Virtualization For DataStores (Work in Progress)
+
+It's a capability to exclude some content from initial snapshot (used when loading container) and thus improve boot
+performance of a container. Excluded content could be loaded at a later time when it's required.
+
+### Motivation for Data Virtualization
+
+This section talks about how the system used to work before Data virtualization. Fluid (through its APIs) does expose
+some levels of data virtualization. But under covers, there is almost none. The whole content of the file is downloaded
+in one go. Due to limitation of data virtualization, FF holds ALL blobs in snapshot as those might be required in the
+future. Any delayed loading (through FF APIs) results in loading state of DDSs at a sequence number of snapshot we
+booted from, and application of all the ops up till current sequence number. While application may choose not to load
+some data stores immediately on boot (and realize some saving in time and memory by not allocating appropriate app
+state for such datastores), FF still pays the costs for such content. It also continues to pay the cost for all such
+content indefinitely, even if those datastores were loaded (this could theoretically be optimized, but now there is
+no good way for driver to learn what blobs are safe to discard due to limitations in interface and lack of appropriate
+semantics at driver API layer).
+
+### Improvement with Data Virtualization
+
+With this, we will provide a capability to:
+
+- Exclude some sub-trees from snapshot payload, thus allowing faster transfer times / boot times and smaller initial
+memory footprint.
+- Ability to delay-load data stores later.
+
+Container Runtime Apis like IContainerRuntimeBase.createDataStore and IContainerRuntimeBase.createDetachedDataStore
+provides an argument `loadingGroupId` which allows apps to mark a datastore at time of creation currently. Every data
+store is assigned a groupID. Not providing groupID (on API) means that default ID is used. This groupId represents the
+group of the datastore within a container or its snapshot. When a container is loaded initially, only datastores which
```

Comment:
Yes, the app should only provide groupid to a datastore if it can get fairly big in size which we mentioned here in the readme. App should not provide groupid to just any datastore. It is ok if all datastores are in 1 group if the content in them is not supposed to be humongous like a subpage. This is what happens today as in all of the are in same group.

Reply:
discussed on chat. Thanks for updating.